### ISSUE TITLE
[BUG] INFO flag raises TypeError

### ISSUE THREAD

**Issue by hylkedonker** [bihealth/vcfpy#146]
* vcfpy version: `master` branch
* Python version: 3.9.7
* Operating System: Ubuntu 21.10

### Description

I have VCF files that generate a 
```
TypeError: argument of type 'bool' is not iterable
```
when reading the lines. Closer inspection reveals that the problem is in the `INFO` field. More specifically, when there is a flag in the INFO field, but the header specifies otherwise, a `TypeError` is generated in `convert_field_value(type_, value)`.

### What I Did
Here is a minimum reproducing example:
```python
import io
import vcfpy

string_buffer = io.StringIO(r"""##fileformat=VCFv4.2
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##INFO=<ID=MH,Number=1,Type=String,Description="Microhomology">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	FOO	BAR
1	4798729	.	CT	C	621	PASS	MH	GT	0/0:25	0/1:64
""")

with vcfpy.Reader.from_stream(string_buffer) as reader:
    for line in reader:
        print(line)
```

### Solution
I think a quick solution would be to adjust `parse_field_value` to detect that a flag, instead of key-value pair, is passed and override behaviour specified in the header.
I can make a pull request.

**Comment by hylkedonker** [comment]
I've made a pull request to fix this problem: https://github.com/bihealth/vcfpy/pull/147

**Comment by holtgrewe** [comment]
Thanks.

**Comment by holtgrewe** [comment]
Released as v0.13.4
